### PR TITLE
Use snapshots on acceptance tests

### PR DESCRIPTION
**Pull request by Tito-Kati**
I have refactored all the acceptance tests to use snapshots, and I have also taken the opportunity to categorize them and add extra tests.

During this process, I found some functionalities that are not working as expected, so I have taken the liberty of marking those tests as incomplete so that we can finish implementing/fixing/etc. them.

## ✅ To-do list

- [x] I updated the `CHANGELOG.md` to reflect the new feature or fix
- [x] I updated the documentation to reflect the changes

### CODE REVIEW

**Comment by Tito-Kati** [comment]
@khru @Chemaclass Can you validate if you like the new acceptance tests? If so, I'll refactor the rest

**Review by Chemaclass — APPROVED** [review]
Wow, nice!

File: tests/acceptance/fixtures/test_bashunit_when_a_execution_error.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+function test_error() {
+  invalid_function_name
+  assert_general_error
+}
```
Comment (khru) [inline tests/acceptance/fixtures/test_bashunit_when_a_execution_error.sh:6]:
```suggestion
function test_error() {
  invalid_function_name
  assert_general_error
}

function test_error_with_double_asserts() {
  invalid_function_name
  assert_general_error
  invalid_function_name_two
  assert_general_error
}
```

Comment (khru) [reply tests/acceptance/fixtures/test_bashunit_when_a_execution_error.sh:6]: I think this could help

**Comment by khru** [comment]
@Tito-Kati this test fails on my machine.

```
✗ Failed: Install downloads the given version
    Expected 'bashunit - 0.8.0'
    but got ' _               _                   _
| |__   __ _ ___| |__  __ __ ____ (_) |_
| '_ \ / _' / __| '_ \| | | | '_ \| | __|
| |_) | (_| \__ \ | | | |_| | | | | | |_
|_.__/ \__,_|___/_| |_|\___/|_| |_|_|\__|
0.8.0'

```

**Comment by Tito-Kati** [comment]
> @Tito-Kati this test fails on my machine.
> 
> ```
> ✗ Failed: Install downloads the given version
>     Expected 'bashunit - 0.8.0'
>     but got ' _               _                   _
> | |__   __ _ ___| |__  __ __ ____ (_) |_
> | '_ \ / _' / __| '_ \| | | | '_ \| | __|
> | |_) | (_| \__ \ | | | |_| | | | | | |_
> |_.__/ \__,_|___/_| |_|\___/|_| |_|_|\__|
> 0.8.0'
> ```
> 
> And the test is `todo`
> 
> ```
> function test_install_downloads_the_given_version() {
>   local install_dir="./lib/bashunit"
>   local output
> 
>   output="$(./install.sh lib 0.8.0)"
> 
>   assert_equals\
>     "$(printf "> Downloading a concrete version: '0.8.0'\n> bashunit has been installed in the 'lib' folder")"\
>     "$output"
>   assert_file_exists "$install_dir"
>   todo "Fix next assertion fails when HEADER_ASCII_ART=true is set in your .env (overriding given --env option)"
>   assert_equals\
>     "$(printf "\e[1m\e[32mbashunit\e[0m - 0.8.0")"\
>     "$("$install_dir" --env "$TEST_ENV_FILE" --version)"
> }
> ```
> 
> Local solution but still a bug: `change my .env config` the test is using my `.env`config and not skipping even with a todo
> 
> image
> ## How to reproduce it
> 
> Change on your local `.env` file
> 
> ```
> HEADER_ASCII_ART=true
> ```
> 
> Then you will see the error

I left the test running because if you don't have the ascii header it works fine, but I added the todo to remind us to fix it (todos don't stop the execution of the tests)

**Pull request merged**